account_reversal/models/account_move.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Copyright 2011 Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2011 Nicolas Bessi (Camptocamp)
+# Copyright 2012-2013 Guewen Baconnier (Camptocamp)
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    to_be_reversed = fields.Boolean(
+        string="To Be Reversed",
+        help="Check this box if your entry has to be reversed at the end "
+             "of period.")
+    reversal_id = fields.Many2one(
+        comodel_name='account.move', ondelete='set null', readonly=True,
+        string="Reversal Entry")
+
+    def _move_lines_reverse_prepare(self, lines, date=False, journal=False,
+                                    line_prefix=False):
+        for line in lines:
```

Comment:
Well, there are several problems in this prepare design, like the fact that you don't know here the source account.move.line, so in practice, there's no use for this method. Remove it or refactor it discarding copy_data dictionary and iterating over source account.move.line for getting the sub-records.